### ISSUE TITLE
Fix docker

### ISSUE THREAD

**Issue by xmunoz** [DARIAEngineering/dcaf_case_management#2606]
- Add css and js hot-reloading capability to dockerized app. This capability never worked after the webpacker exorcism, and was removed in PR #2605. We should really get this working.
- Add docker build back to CI
- Update `docs/SETUP.md` with the docker build instructions

**Comment by lomky** [comment]
Found the issue after an hour of debugging with @rocketnova. 

Our build:css command invokes the sass --watch option, however:

https://sass-lang.com/documentation/cli/dart-sass#watch
> Sass watches only the directories that you pass as-is on the command line, parent directories of filenames you pass on the command line, and load paths. It does not watch additional directories based on a file’s @import/@use/ @forward rules.

we are only watching the collected top level file `app/assets/stylesheets/application.bootstrap.scss` - which just imports many other files. The initial sass load does follow the imports, but the watch, as stated above, does not watch anything other than that file directly. You can prove this to yourself by changing a value in the lower files (no reload) and directly in that file (reloads!) 

TLDR: We need a distinct command for watching all the scss files, or to restructure somehow to enable this? I'll keep debugging this later unless you know the fix immediately @colinxfleming - I'm not familiar enough with sass to know how to do this immediately

**Comment by lomky** [comment]
Current State:
 - dev env works (our README incorrectly says it doesnt) (fixed in #2620, tweaked for windows in #2636)
 - css hotreloading is known broken, see above.
 - js hot reloading is unclear if functional (I haven't checked)
 - docker CI test is still broken (commented out in #2620 to prioritize dev env)

**Comment by xmunoz** [comment]
The dev env was not working for me, and if it's not working in CI, then I don't we can definitively say that it's working.

**Comment by lomky** [comment]
Found and fixed the JS issue - the container needed the path to esbuild. This also seems to have fixed the dev env and CI build.

I am rather perplexed at the sass hot reloading issue. As far as I can tell the default configuration suggested by cssbundling-rails will never have working hot reloading for sass and a `./app/assets/stylesheets/application.sass.scss` that consists of `@import` statements. But I have not been able to find anyone else reporting this issue, which makes me suspect I'm still missing something.

**Comment by xmunoz** [comment]
Yeah, I'm not sure that hot reloading ever worked. This is why I wanted to get some documentation in this repo about our new asset pipeline, but I'm not even sure who would write it since none of us know how it works.

I do think this dovetails nicely with a larger issue though, which is that we don't have any senior front-end expertise on the core team. Maybe something to discuss at the next board meeting.

**Comment by colinxfleming** [comment]
I don't think there's a specific docker problem to fix at this point (and the broader frontend issues are ticketed elsewhere), so going to close this out
